cmd/atlas/internal/mux/mux.go

```diff
@@ -0,0 +1,72 @@
+package mux
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"ariga.io/atlas/sql/schema"
+)
+
+type (
+	// Mux is used for routing dsn to correct provider.
+	Mux struct {
+		providers map[string]func(string) (*Driver, error)
+	}
+
+	// Driver implements the Atlas interface.
+	Driver struct {
+		*sql.DB
+		Differ    schema.Differ
+		Execer    schema.Execer
+		Inspector schema.Inspector
+	}
+)
+
+// Close release all resources taken by Driver.
+func (a *Driver) Close() error {
+	return a.DB.Close()
+}
+
+// NewMux returns a new Mux.
+func NewMux() *Mux {
+	return &Mux{
+		providers: make(map[string]func(string) (*Driver, error)),
+	}
+}
+
+var defaultURLMux = NewMux()
+
+// DefaultMux returns the default system "Mux".
+func DefaultMux() *Mux {
+	return defaultURLMux
+}
```

Comment:
See `http.DefaultClient` as an example.

Reply:
so to remove it that users won't get confused and think they get a default configuration?